src/lambda/commands/downloadLambda.ts

```diff
@@ -94,7 +95,9 @@ async function runDownloadLambda(
 
             try {
                 lambdaLocation = path.join(downloadLocation, getLambdaDetails(functionNode.configuration).fileName)
-                await downloadAndUnzipLambda(progress, functionNode, downloadLocation)
+                const lambdaFunction = await downloadAndUnzipLambda(progress, functionNode, downloadLocation)
+                const appJson = createApplicationJson(lambdaFunction, functionNode.regionCode)
+                writeAppJsonFile(downloadLocation, appJson)
```

Comment:
createApplicationJson() could also write the file. The the file could be read by the test. That makes the test slightly more useful and meaningful: it checks path issues, encoding issues, filesystem failures, etc.

Reply:
That makes sense, it instead would test the final result rather than the step before the write. Updated in the last commit.